### ISSUE TITLE
AttributeError: 'SimpleAsyncOperation' object has no attribute 'thread'

### ISSUE THREAD

**Issue by mexicarne** [zim-desktop-wiki/zim-desktop-wiki#647]
I'm trying zim 0.70-rc2 on FreeBSD 13 with python 3.7.2. I got two exceptions on first run of 0.70 with my old notebooks:

1. This one has disappeared upon second run:
```python
ERROR: Exception while loading plugin: calendar
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/zim/plugins/__init__.py", line 361, in load_plugins_from_preferences
    self.load_plugin(name)
  File "/usr/local/lib/python3.7/site-packages/zim/plugins/__init__.py", line 459, in load_plugin
    klass = self.get_plugin_class(name)
  File "/usr/local/lib/python3.7/site-packages/zim/plugins/__init__.py", line 413, in get_plugin_class
    mod = get_module(modname)
  File "/usr/local/lib/python3.7/site-packages/zim/utils.py", line 35, in get_module
    mod = __import__(name)
ModuleNotFoundError: No module named 'zim.plugins.calendar'
```

2. But this one still persist every time I'm run zim --verbose:
```python
INFO: Running: ('git', 'status', '--porcelain') (cwd: /usr/home/user/Notebooks/Notes)
ERROR: Failed loading extension <class 'zim.plugins.versioncontrol.VersionControlMainWindowExtension'> for plugin <zim.plugins.versioncontrol.VersionControlPlugin object at 0x8043a6518>
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/zim/plugins/__init__.py", line 437, in _extend
    ext = ext_class(plugin, obj)
  File "/usr/local/lib/python3.7/site-packages/zim/plugins/versioncontrol/__init__.py", line 130, in __init__
    self.on_preferences_changed(None, start=True)
  File "/usr/local/lib/python3.7/site-packages/zim/plugins/versioncontrol/__init__.py", line 147, in on_preferences_changed
    self.do_save_version_async()
  File "/usr/local/lib/python3.7/site-packages/zim/plugins/versioncontrol/__init__.py", line 177, in do_save_version_async
    self._autosave_thread = op.thread
AttributeError: 'SimpleAsyncOperation' object has no attribute 'thread'
```
Is there something to do with this one? Thanks

**Comment by jaap-karssenberg** [comment]
The first item is not a bug, the "calendar" plugin got replaced with "journal". Only will warn first time you use them.

2nd item I will fix in a minute

**Comment by mexicarne** [comment]
This commit fixed original problem, but it looks like new one was introduced:
```python
INFO: Running: ('git', 'status', '--porcelain') (cwd: /usr/home/user/Notebooks/Notes)
INFO: Running: ('git', 'add', '/usr/home/user/Notebooks/Notes') (cwd: /usr/home/user/Notebooks/Notes)
INFO: Running: ('git', 'status', '--porcelain') (cwd: /usr/home/user/Notebooks/Notes)
INFO: Running: ('git', 'commit', '-m', 'Automatically saved version from zim') (cwd: /usr/home/user/Notebooks/Notes)
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/zim/notebook/operations.py", line 262, in __iter__
    progress = next(self._do_work)
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/zim/notebook/index/__init__.py", line 369, in <lambda>
    GObject.idle_add(lambda: next(my_iter, False), priority=GObject.PRIORITY_LOW)
  File "/usr/local/lib/python3.7/site-packages/zim/notebook/index/__init__.py", line 383, in on_idle_iter
    self.callback()
  File "/usr/local/lib/python3.7/site-packages/zim/notebook/index/__init__.py", line 170, in <lambda>
    self.background_check.callback = lambda *a: on_out_of_date_found(notebook, self.background_check)
  File "/usr/local/lib/python3.7/site-packages/zim/notebook/index/__init__.py", line 401, in on_out_of_date_found
    op.run_on_idle()
  File "/usr/local/lib/python3.7/site-packages/zim/notebook/operations.py", line 232, in run_on_idle
    self.notebook._operation_check() # can raise
  File "/usr/local/lib/python3.7/site-packages/zim/notebook/operations.py", line 309, in __call__
    self._join()
  File "/usr/local/lib/python3.7/site-packages/zim/notebook/operations.py", line 319, in _join
    for i in self:
RuntimeError: generator raised StopIteration
```

**Comment by mexicarne** [comment]
> That one should be addressed already in [6f34126](https://github.com/zim-desktop-wiki/zim-desktop-wiki/commit/6f341264e62fd2da19cfe6a1deca22d6a2bf56c5), if not let me know

Yes, indeed. Thank you!
